src/Generator/Generators/CSharp/CSharpExpressionPrinter.cs

```diff
@@ -68,8 +68,8 @@ public string VisitExpression(ExpressionObsolete expr)
                 case StatementClass.ConstructorReference:
                     var constructorExpr = (CXXConstructExprObsolete)expr;
                     if (constructorExpr.Arguments.Count == 1 &&
-                        constructorExpr.Arguments[0].Declaration is Enumeration.Item)
-                        return constructorExpr.Arguments[0].Declaration.Visit(typePrinter).Type;
+                        constructorExpr.Arguments[0].Class != StatementClass.Any)
```

Comment:
I am blocking `StatementClass.Any` here since such statements will be generated without any kind of modification and may contain code that is invalid in C#.

Otherwise, `defaultMappedToEnumAssignedWithCtor` fails because `nullptr` is not valid C#.

It would be better if CppSharp is aware that `Arguments[0]` has been defaulted away and instead report 0 arguments to avoid this mess altogether.